tokio/src/sync/mutex.rs

```diff
@@ -86,11 +6,89 @@ use std::error::Error;
 use std::fmt;
 use std::ops::{Deref, DerefMut};
 
-/// An asynchronous mutual exclusion primitive useful for protecting shared data
+/// An asynchronous `Mutex`-like type.
 ///
-/// Each mutex has a type parameter (`T`) which represents the data that it is protecting. The data
-/// can only be accessed through the RAII guards returned from `lock`, which
-/// guarantees that the data is only ever accessed when the mutex is locked.
+/// This module provides [`Mutex`], a type that acts similarly to an asynchronous [`std::sync::Mutex`], with one
+/// major difference: the [`MutexGuard`] returned by `lock` is not tied to the lifetime of the
+/// `Mutex`. This enables you to acquire a lock, and then pass that guard into a future, and then
+/// release it at some later point in time.
```

Comment:
This is not true. The difference is that the lock function does not block, and that the guard is `Send`. The guard still borrows from the mutex.

Reply:
addressed the inaccuracies, but probably still needs polishing, can you check again please?